### PR TITLE
Feat/forward prop

### PR DESCRIPTION
**Pull request by devrnt**
### Info
Add `shouldForwardProp` to hide to boilerplate for the `forwardProp` function in `setup`.
The user should pass a filter function that resolves to a boolean. Truthy results of the filter function will be let through,  falsy results will be removed from the `props`.

See https://github.com/cristianbote/goober/pull/227#discussion_r553164138

### Use
```jsx
import * as React from 'react';
import { setup } from 'goober';
import { shouldForwardProp } from 'goober/forwardProp';

setup(
  React.createElement,
  undefined,
  undefined,
  shouldForwardProp((prop) => {
    return prop[0] !== '$';
  }),
);
```

~~Don't mind commit `7bf7ff9` it's in sequence of https://github.com/cristianbote/goober/pull/233~~ 
I did a rebase

### CODE REVIEW

**Comment by cristianbote** [comment]
woow! 😄 this looks great!

File: package.json [hunk 1]
```diff
@@ -49,6 +53,12 @@
       "import": "./prefixer/dist/goober-autoprefixer.mjs",
       "require": "./prefixer/dist/goober-autoprefixer.js",
       "umd": "./prefixer/dist/goober-autoprefixer.umd.js"
+    },
+    "./forwardProp": {
```
Comment (cristianbote) [inline package.json:57]: How do you feel if we rename the `forwardProp` in `shouldForwardProp`? 🤔 it's longer, but might look more familiar for folks coming from other libraries.

Comment (devrnt) [reply package.json:57]: Yeah seems good to me!

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,6752 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (devrnt) [inline yarn.lock:1]: Will drop this `yarn.lock`

File: package.json (hunk 1 above)
Comment (devrnt) [reply package.json:57]: Do you want these reference to be changed everywhere (like in `forwardProp/package.json`) or just this "directory export"?

Comment (cristianbote) [reply package.json:57]: How about everywhere for the sake of consistency? So the name should be camelCase (`shouldForwardProp`) and the folder should be dash-separated(`should-forward-prop`).

File: forwardProp/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  "scripts": {
+    "build:lib": "rm -rf dist && microbundle --entry src/index.js --name gooberForwardProp --no-sourcemap",
+    "test": "jest --setupFiles ./jest.setup.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/cristianbote/goober.git",
+    "directory": "forwardProp"
+  },
+  "author": "Cristian <botecristian@yahoo.com>",
```
Comment (cristianbote) [inline forwardProp/package.json:22]: While this indeed is under my user's scope, this package is all on you @devrnt! 😃  So, I would encourage you to stamp your name in here! 🎉

Comment (devrnt) [reply forwardProp/package.json:22]: Okay changes noted 😀

File: package.json (hunk 1 above)
Comment (devrnt) [reply package.json:57]: Sounds good to me!

File: shouldForwardProp/should-forward-prop.d.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export = gooberGlobal;
```
Comment (cristianbote) [inline shouldForwardProp/should-forward-prop.d.ts:1]: Nit: I think this should be changed to 
```suggestion
export = gooberShouldForwardProp;
```

right?

File: shouldForwardProp/should-forward-prop.d.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+export as namespace shouldForwardProp;
+
+declare namespace gooberGlobal {
```
Comment (cristianbote) [inline shouldForwardProp/should-forward-prop.d.ts:5]:
```suggestion
declare namespace gooberShouldForwardProp {
```

File: package.json [hunk 6]
```diff
@@ -20,6 +20,10 @@
     "prefixer/package.json",
     "prefixer/autoprefixer.d.ts",
     "prefixer/README.md",
+    "shouldForwardProp/src",
```
Comment (cristianbote) [inline package.json:23]: Uh needs the `shouldForwardProp/dist` as well. But I'll take care of it later 👍

**Review by cristianbote — APPROVED** [review]
woop woop! 🙌  awesome!

File: shouldForwardProp/src/index.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+export function shouldForwardProp(filterPropFunction) {
+    return function forwardProp(props) {
+        for (let p in props) {
```
Comment (devrnt) [inline shouldForwardProp/src/index.js:3]: Will call this `prop` for convenience

**Review by cristianbote — APPROVED** [review]

**Pull request merged**